src/core/metrics/app-metrics.ts

```diff
@@ -32,14 +38,41 @@ export class AppMetrics {
   private completed: Record<string, number> = {};
   private eventLog: string[] = [];
   private settings = new SessionSettings<'KONTUR_SQ_ALERT' | 'KONTUR_SQ_LOG'>();
+  reportTemplate: {
+    name: string;
+    value: number;
+    type: string;
+    appId: string;
+    userId: string | null;
+    buildVersion: string;
+  } = {
+    name: '', // metric name, e.g. full-load-time, map-load-time, disasters-panel-load-time, etc.
+    value: 0, // time to full load in ms
+    type: 'SUMMARY',
+    appId: '',
+    userId: '', // null if user is not authenticated
+    buildVersion: `${import.meta.env.PACKAGE_VERSION}-${import.meta.env.MODE}`,
+  };
+  listener: void;
 
   constructor() {
     globalThis.KONTUR_METRICS = {
+      watchList: this.watchList,
+      markers: this.markers,
       sequences: this.sequences,
       events: this.eventLog,
       toggleAlert: () => this.settings.toggle('KONTUR_SQ_ALERT'),
       toggleLog: () => this.settings.toggle('KONTUR_SQ_LOG'),
     };
+    this.listener = globalThis.addEventListener(METRICS_EVENT, ((e: MetricsEvent) => {
+      this.processEvent(e.detail.name, e.detail.payload);
+    }) as EventListener);
```

Comment:
should that be a TODO to extend global events for our app with our custom event?
typecasting sometimes can be dangereous

Reply:
current types for CustomEvent has issues, this is easiest way ATM